src/seedu/addressbook/data/person/Name.java

```diff
@@ -37,6 +37,10 @@ public static boolean isValidName(String test) {
         return test.matches(NAME_VALIDATION_REGEX);
     }
 
+    public boolean isSimilar(Name other){
```

Comment:
you can add the "unusual" cases like `null` test first before jumping in to a minimal valid case.